pvlib/clearsky.py

```diff
@@ -765,10 +797,17 @@ def detect_clearsky(measured, clearsky, times=None, window_length=10,
     else:
         meas = measured
 
-    if not isinstance(clearsky, pd.Series):
-        clear = pd.Series(clearsky, index=times)
+    if not isinstance(clear_sky, pd.Series):
+        clear = pd.Series(clear_sky, index=times)
+    # This clause is designed to address cases where measured has missing time
+    # steps - if this is the case, clear should be set to have the same
+    # missing time intervals as measured. Not doing this may cause issues with
+    # arrays of different lengths when evaluating comparison criteria and
+    # when indexing the Hankel matrix to construct clear_samples
+    elif len(clear_sky.index) != len(times):
+        clear = pd.Series(clear_sky, index=times)
```

Comment:
Untested, and I suspect this will fail - `clear_sky` is a `Series` of different length than `times` so it can't get indexed by `times`.

Reply:
Just added a test for this in test_clearsky.py